solr/core/src/test/org/apache/solr/handler/ReplicationTestHelper.java

```diff
@@ -0,0 +1,317 @@
+            log.info("Waited for {}ms and found {} docs", timeSlept, numFound(res));
+        }
+        return res;
+    }
+
+    public static long numFound(NamedList<Object> res) {
+        return ((SolrDocumentList) res.get("response")).getNumFound();
+    }
+
+    public static NamedList<Object> getDetails(SolrClient s) throws Exception {
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.set("command","details");
+        params.set("_trace","getDetails");
+        params.set("qt",ReplicationHandler.PATH);
+        QueryRequest req = new QueryRequest(params);
+
+        NamedList<Object> res = s.request(req);
+        assertReplicationResponseSucceeded(res);
+
+        @SuppressWarnings("unchecked") NamedList<Object> details
+                = (NamedList<Object>) res.get("details");
+
+        assertNotNull("null details", details);
+
+        return details;
+    }
+
+    public static NamedList<Object> getIndexVersion(SolrClient s) throws Exception {
+
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.set("command","indexversion");
+        params.set("_trace","getIndexVersion");
+        params.set("qt",ReplicationHandler.PATH);
+        QueryRequest req = new QueryRequest(params);
+
+        NamedList<Object> res = s.request(req);
+        assertReplicationResponseSucceeded(res);
+
+        return res;
+    }
+
+    public static void assertReplicationResponseSucceeded(NamedList<?> response) {
+        assertNotNull("null response from server", response);
+        assertNotNull("Expected replication response to have 'status' field", response.get("status"));
+        assertEquals("OK", response.get("status"));
+    }
+
+    public static NamedList<Object> reloadCore(SolrClient s, String core) throws Exception {
+
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.set("action","reload");
+        params.set("core", core);
+        params.set("qt","/admin/cores");
+        QueryRequest req = new QueryRequest(params);
+
+        try (HttpSolrClient adminClient = adminClient(s)) {
+            NamedList<Object> res = adminClient.request(req);
+            assertNotNull("null response from server", res);
+            return res;
+        }
+    }
+
+    public static HttpSolrClient adminClient(SolrClient client) {
+        String adminUrl = ((HttpSolrClient)client).getBaseURL().replace("/collection1", "");
+        return SolrTestCaseJ4.getHttpSolrClient(adminUrl);
+    }
+
+
+    public static void pullFromTo(String srcUrl, String destUrl) throws IOException {
+        URL url;
+        InputStream stream;
+        String leaderUrl = destUrl
+                + ReplicationHandler.PATH+"?wait=true&command=fetchindex&leaderUrl="
+                + srcUrl
+                + ReplicationHandler.PATH;
+        url = new URL(leaderUrl);
+        stream = url.openStream();
+        stream.close();
+    }
+
+    public static class SolrInstance {
```

Comment:
I'm not clear what value this provides, can you elaborate a little bit?

Reply:
I see that this was copied from another location, so maybe this doesn't need to be addressed now.